### DISCUSSION TITLE
Is it possible to create stubs from libraries "built in" to a custom micropython?

### DISCUSSION THREAD

**Discussion by trepidacious** [BradenM/micropy-cli#402] [Q&A] [answered]
[Pimoroni](https://shop.pimoroni.com) provide a [custom micropython build](https://github.com/pimoroni/pimoroni-pico#micropython) that includes a number of additional libraries for their Raspberry Pi Pico add-ons and RP2040-based boards. I've successfully generated stubs from a Pico board with this version of micropython installed, however it only seems to have the standard libraries (e.g. `machine` etc.), and not some of the custom libraries (e.g. [`plasma`](https://github.com/pimoroni/pimoroni-pico/tree/main/micropython/modules/plasma) for driving LED strips).

Should the stubs for these libraries already be generated, or is there something I can do to generate them? If there's something Pimoroni could do to provide for stub generation I could raise this in their forum, at the moment I don't know enough about the structure of micropython to guess what might be needed!

**Comment by BradenM** [comment, marked as answer]
@trepidacious
I have been working alongside @Josverl to bring this exact functionality to light :smile: 

See:
https://github.com/Josverl/micropython-stubber/pull/298
https://github.com/Josverl/micropython-stubber/pull/319

I intend to integrate this feature into micropy-cli in #394 (or possibly right after).

**Reply by BradenM** [reply]
@trepidacious 
Hoping to bring this in a v4.2.0-beta1 release next (see: https://github.com/BradenM/micropy-cli/pull/422) where it snuck in here: https://github.com/BradenM/micropy-cli/pull/422/files#diff-0bb101fc9490dbbf83340f0c97ab833219488f279f811d6bfc3991e7d4f2ab5bR79-R82
